### PR TITLE
[express-graphql] Fixed the definition of some optional fields.

### PR DESCRIPTION
**Pull request by firede**
...and update tests, that would cause the new version `@types/graphql` test to fail. #24566

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/graphql/express-graphql/blob/master/src/index.js>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/express-graphql/index.d.ts [hunk 1]
```diff
@@ -80,32 +84,32 @@ declare namespace graphqlHTTP {
...
-        operationName?: string;
-        raw?: boolean;
+        query: string | null | undefined;
+        variables: { [name: string]: any } | null | undefined;
+        operationName: string | null | undefined;
+        raw: boolean | null | undefined;
     }
 
-    type Middleware = (request: Request, response: Response) => void;
+    type Middleware = (request: Request, response: Response) => Promise<undefined>;
```
Comment (Hubel) [inline types/express-graphql/index.d.ts]: In https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L34 the RequestHandler is defined as
`(req: Request, res: Response, next: NextFunction): any;`
If this line has to be changed, I would vote for `any` as result type.

**Review by people-ware — CHANGES_REQUESTED** [review]
Changing the Middleware result to an explicit type might introduce unnecessary trouble

File: types/express-graphql/index.d.ts (hunk 1 above)
Comment (firede) [reply types/express-graphql/index.d.ts]: thanks for review.
In [express-graphql](https://github.com/graphql/express-graphql/blob/v0.6.12/src/index.js#L123) the `Middleware` returns `Promise<void>`, compatible with `RequestHandler`'s return value.
Of course `any` is safe, but should we follow the definition of `express-graphql`?

Comment (Hubel) [reply types/express-graphql/index.d.ts]: You're right. The definition should stick to the graphql-express rather than the general express types.

**Review by people-ware — APPROVED** [review]

**Pull request merged**